### PR TITLE
Force refresh client on update

### PR DESCRIPTION
**Pull request by eduardb**
Clients were not refreshing anymore after updates, because the '/refresh' endpoint that was hit by the CI on updates was removed from the server.

We found that instead we can force the refresh on the client when it loses the connection to the server, because most likely that would happen when the server is restarted after being updated.

Paired with @florianmski

### CODE REVIEW

File: angular2-client/src/app/dashboard-carousel/socket.service.ts [hunk 1]
```diff
@@ -25,9 +25,15 @@ export class SocketService {
     const observable: Observable<any> = Observable.create((observer: Observer<any>) => {
       // The Observable will not complete or fail since we want to reconnect if possible.
       socket.on('message', observer.next.bind(observer));
-      socket.on('refresh', observer.next.bind(observer, { payload: {}, widgetKey: 'refresh' }));
```
Comment (eduardb) [inline angular2-client/src/app/dashboard-carousel/socket.service.ts]: This message is not sent by the server anymore

**Comment by eduardb** [comment]
Changing implementation details 😂

File: angular2-client/src/app/dashboard-carousel/socket.service.ts [hunk 2]
```diff
@@ -25,9 +25,15 @@ export class SocketService {
...
+      // When the connection to the server is lost, most likely we are updating it,
+      // so we should refresh the client to have the latest changes
+      socket.on('disconnect', () => this.forceRefreshClient(observer));
```
Comment (frapontillo) [inline angular2-client/src/app/dashboard-carousel/socket.service.ts]: Do we want to add a max number of retries?

Comment (eduardb) [inline angular2-client/src/app/dashboard-carousel/socket.service.ts]: That is already defined above 👅 
https://github.com/novoda/dashboards/pull/39/files/dc5762df9ce234c84da1cf1d9a24d48f610fd0c2#diff-b4d0c1e42b7679213b3cdbd9eb650a41L21

**Comment by eduardb** [comment]
@frapontillo :shipit:

**Pull request merged**